### ISSUE TITLE
`set soft-wrap.enable true` should be set by default in `.md` files

### ISSUE THREAD

**Issue by tapyu** [helix-editor/helix#8331]
I suppose that it is sensible to set soft-wrap feature to true in the default settings for [markdown][1] files as it often involves long paragraphs....

[1]: https://github.com/helix-editor/helix/blob/master/languages.toml#L1239

**Comment by archseer** [comment]
Agreed but that config option is currently global. It'll require some work to introduce document specific/language specific config.

**Comment by tapyu** [comment]
@archseer I am really sorry, but what you mean by "being global"? Can't we set this config for markdown files? For instance:

```md
[[language]]
name = "markdown"
scope = "source.md"
injection-regex = "md|markdown"
file-types = ["md", "markdown", "PULLREQ_EDITMSG"]
soft-wrap.enable = true
roots = [".marksman.toml"]
language-servers = [ "marksman" ]
indent = { tab-width = 2, unit = "  " }
```

??

**Comment by kirawi** [comment]
No. By global it means that it's only in `config.toml`

**Comment by gabydd** [comment]
In the case of softwrap it can be set per language, I don't think it is in the docs but it does work and is defined here: https://github.com/helix-editor/helix/blob/1c88432efc3724f60b27d580b8b490040a3048f6/helix-core/src/syntax.rs#L107

**Comment by tapyu** [comment]
I also confirm that the `soft-wrap` feature can be set to a specific language. At least, it properly works for the `~/.config/helix/languages.toml` file

```
❯ cat languages.toml
[[language]]
name = "bash"
# scope = "source.sh" # it is already set on the default language settings
# file-types = ["sh", "zsh"] # it is already set on the default language settings
# comment-token = "#" # it is already set on the default language settings
language-server = { command = "bash-language-server", args = ["start"] }


[[language]]
name = "markdown"
soft-wrap.enable = true
```

**Comment by archseer** [comment]
Hmm seems that this isn't documented https://github.com/helix-editor/helix/blob/1c88432efc3724f60b27d580b8b490040a3048f6/book/src/languages.md?plain=1#L53-L70

**Comment by arcstur** [comment]
Since the PR was merged I think this can be closed

**Comment by the-mikedavis** [comment]
That PR added the docs for `soft-wrap` in the `languages.toml` config but this issue is about adding `soft-wrap = true` for the markdown block in the default `languages.toml` config in this repo: https://github.com/helix-editor/helix/blob/c3cb1795bfd2bfe1c01badd71ac101b0c8fdd379/languages.toml#L1328-L1335

**Comment by arcstur** [comment]
Oh, you're right, haha. So, are we going to add that? If yes I can make the PR.

**Comment by the-mikedavis** [comment]
Yep feel free to send a PR

**Comment by wmstack** [comment]
Also text files? This is how it is in Zed.

**Comment by tapyu** [comment]
It is also sensible to set it to `.txt` files, for sure.

**Comment by wmstack** [comment]
Would that work for empty scratch buffers by default as well? Are they assumed to be text based.
